bin/node/src/service.rs

```diff
@@ -375,51 +302,104 @@ pub fn new_authority(
 
                 Ok((slot, timestamp))
             },
-            force_authoring,
+            force_authoring: config.force_authoring,
             backoff_authoring_blocks,
-            keystore: keystore_container.local_keystore(),
+            keystore: service_components.keystore_container.local_keystore(),
             sync_oracle: sync_oracle.clone(),
             justification_sync_link: (),
             block_proposal_slot_portion: SlotProportion::new(2f32 / 3f32),
             max_block_proposal_slot_portion: None,
-            telemetry: telemetry.as_ref().map(|x| x.handle()),
+            telemetry: service_components.other.1.as_ref().map(|x| x.handle()),
```

Comment:
Stuff like `service_components.other.1` is quite ugly. As far as I can tell there is no real reason to use `PartialComponents` and we could just write our own config/components struct that has more descriptive names for everything we need – at least I think so, there seem to be no uses of this struct from functions that are not ours. Are you planning to do something like this as a continuation of this refactor?

Reply:
YOu are 100% right about PartialComponents. I wanted to do it in this part but I forgot to do it. Now, I think its best to have it in the continuation :P